freqtrade/strategy/backtest_lookahead_bias_checker.py

```diff
@@ -0,0 +1,252 @@
+        timestamp = int(dt.replace(tzinfo=timezone.utc).timestamp())
+        return timestamp
+
+    @staticmethod
+    def get_result(backtesting, processed):
+        min_date, max_date = get_timerange(processed)
+
+        result = backtesting.backtest(
+            processed=deepcopy(processed),
+            start_date=min_date,
+            end_date=max_date
+        )
+        return result
+
+    @staticmethod
+    def report_signal(result, column_name, checked_timestamp):
+        df = result['results']
+        row_count = df[column_name].shape[0]
+
+        if row_count == 0:
+            return False
+        else:
+
+            df_cut = df[(df[column_name] == checked_timestamp)]
+            if df_cut[column_name].shape[0] == 0:
+                # print("did NOT find the same signal in column " + column_name +
+                #       " at timestamp " + str(checked_timestamp))
+                return False
+            else:
+                return True
+        return False
+
+    # analyzes two data frames with processed indicators and shows differences between them.
+    def analyze_indicators(self, full_vars, cut_vars, current_pair):
+        # extract dataframes
+        cut_df = cut_vars.indicators[current_pair]
+        full_df = full_vars.indicators[current_pair]
+
+        # cut longer dataframe to length of the shorter
+        full_df_cut = full_df[
+            (full_df.date == cut_vars.compared_dt)
+        ].reset_index(drop=True)
+        cut_df_cut = cut_df[
+            (cut_df.date == cut_vars.compared_dt)
+        ].reset_index(drop=True)
+
+        # compare dataframes
+        if full_df_cut.shape[0] != 0:
+            if cut_df_cut.shape[0] != 0:
+                compare_df = full_df_cut.compare(cut_df_cut)
+
+                if compare_df.shape[0] > 0:
+                    for col_name, values in compare_df.items():
+                        col_idx = compare_df.columns.get_loc(col_name)
+                        compare_df_row = compare_df.iloc[0]
+                        # compare_df now comprises tuples with [1] having either 'self' or 'other'
+                        if 'other' in col_name[1]:
+                            continue
+                        self_value = compare_df_row[col_idx]
+                        other_value = compare_df_row[col_idx + 1]
+
+                        # output differences
+                        if self_value != other_value:
+
+                            if not self.current_analysis.false_indicators.__contains__(col_name[0]):
+                                self.current_analysis.false_indicators.append(col_name[0])
+                                print(f"=> found look ahead bias in indicator {col_name[0]}. " +
+                                      f"{str(self_value)} != {str(other_value)}")
+
+    def prepare_data(self, varHolder, pairs_to_load):
+
+        # purge previous data
+        abs_folder_path = pathlib.Path("user_data/models/uniqe-id").resolve()
```

Comment:
`uniqe-id` is assumed to be the id used ? 

If we assume that's the "backtest bias reporter" - then we should also explicitly set this.
otherwise it's taking the assumption that `"identifier": "uniqe-id",` is set - which we shouldn't hard-code or assume here.

We should also (probably not just here) assume that `Path('user_data')` is the user-data.
i can set this to `--userdir ../my_user_data` - which should be respected / used.

Reply:
unique-id was a solution after a talk with drafty / your_secret_admirer on the freqai discord channel.
He suggested that solution. Here a link of the conversation:
https://discord.com/channels/700048804539400213/983285489299779645/1097244068561555607

If you have any better solution just say so, purging unique-id feels wrong to me but he said it's ok so ...

Reply:
my point is - we're taking 2 assumptions here.
* user-data is in `./user_data` (starting from current dir) - which it may or may not be.
* identifier is set to "unique-id"

The first is a (easily solved) problem. 
now for the 2nd, i'm not sure what the intend is (also not from the linked conversation) 

Are you trying to (intentionally) retrain models for each backtest? (i assume so - as the timerange will be different each time??)
if yes - then using a hardcoded identifier may work - but only under the assumption that you assign that same value to the config (which last time - i didn't see).

Reply:
> Are you trying to (intentionally) retrain models for each backtest? (i assume so - as the timerange will be different each time, and we'd like to start fresh).
yes, throw away anything old

Might be better that we force the user to define some ID they want instead of "unique-id"

Reply:
using the config.freqai.identifier now

It looks like it defaults to "example" and still doesn't crash if the user doesn't define it.
The program will still try to purge the previous folder if exists.